cardano-api/src/Cardano/Api/Value.hs

```diff
@@ -115,6 +131,98 @@ instance Semigroup Value where
 instance Monoid Value where
   mempty = Value Map.empty
 
+instance ToJSON Value where
+  toJSON = toJSON . unflatten
+
+unflatten :: Value -> AssetsBundle
+unflatten (Value flatMap) = AssetsBundle $ Map.foldlWithKey folder [] flatMap
+ where
+  folder
+    :: [AssetIdBundle]
+    -> AssetId
+    -> Quantity
+    -> [AssetIdBundle]
+  folder acc (AssetId pid aName) q = AssetIdBundle pid (Map.singleton aName q) : acc
+  folder acc AdaAssetId q = AdaAsset q : acc
```

Comment:
I think this would also be clearer in a bluk style rather than a one-by-one foldl style.